packages/grafana-data/src/types/data.ts

```diff
@@ -107,6 +107,12 @@ export interface QueryResultMeta {
   limit?: number; // used by log models and loki
   json?: boolean; // used to keep track of old json doc values
   instant?: boolean;
+
+  /**
+   * Array of field indices which values create a unique id for each row. Ideally this should be globally unique ID
+   * but that isn't guarantied.
+   */
+  uniqueId?: number[];
```

Comment:
Interesting -- this took me a while to figure out what the intention is.  I like it, but think we should change the names so the property does not seem like it is (itself) a uniqueId 🤔 

In SQL we have: https://github.com/grafana/grafana/blob/v10.2.3/pkg/services/sqlstore/migrations/dashboard_mig.go#L176 
```
{Cols: []string{"dashboard_id", "term"}, Type: UniqueIndex},
```

maybe named "uniqueKeyIndex"?   

can we add the same thing to https://github.com/grafana/grafana-plugin-sdk-go/blob/main/data/frame_meta.go#L55

Reply:
@leeoniya suggested `keyBy` which to me on the other hand didn't suggest at all what was but can see the other point so am open to suggestions. Would prefer something that has `uniqueId` or `uniqueKey` in it because that right away suggest the context. `uniqueKeyFields` or `uniqueIdFields` to me would be preferrable as it has `uniqueId` in it, suffix `Fields` suggest it references fields somehow and with type being `number[]` should be easy extrapolation that it is probably field indexes.

Reply:
How about `uniqueRowIdFields: number[]` .

> Indices of fields for which their combined values form a unique identifier for each row within a frame.

This attempts to emphasizes the unique identifier is for rows.

Reply:
Yep that looks good to me

Reply:
@ryantxu added also here https://github.com/grafana/grafana-plugin-sdk-go/pull/865